auth/service.go

```diff
@@ -186,14 +186,7 @@ func (svc service) Authorize(ctx context.Context, pr PolicyReq) error {
 func (svc service) checkPolicy(ctx context.Context, pr PolicyReq) error {
 	// Domain status is required for if user sent authorization request on things, channels, groups and domains
 	if pr.SubjectType == UserType && (pr.ObjectType == GroupType || pr.ObjectType == ThingType || pr.ObjectType == DomainType) {
-		domainID := pr.Domain
-		if domainID == "" {
-			if pr.ObjectType != DomainType {
-				return svcerr.ErrDomainAuthorization
-			}
-			domainID = pr.Object
-		}
-		if err := svc.checkDomain(ctx, pr.SubjectType, pr.Subject, domainID); err != nil {
+		if err := svc.checkDomain(ctx, pr.SubjectType, pr.Subject, pr.Domain); err != nil {
```

Comment:
@WashingtonKK , 
Please revert back to original, My suggestion doesn't works properly for domain related operations 
It fails at domain authroization